cmd/ddev/cmd/config.go

```diff
@@ -33,22 +47,46 @@ var ConfigCommand = &cobra.Command{
 
 		c, err := ddevapp.NewConfig(appRoot, provider)
 		if err != nil {
-			// If there is an error reading the config and the file exists, we're not sure
-			// how to proceed.
-			if c.ConfigExists() {
-				util.Failed("Could not read config: %v", err)
-			}
+			util.Failed("Could not create new config: %v", err)
 		}
 
 		// Set the provider value after load so we can ensure we use the passed in provider value
 		// for this configuration.
 		c.Provider = provider
+		c.Name = siteName
+		c.Docroot = docrootRelPath
 
-		err = c.Config()
-		if err != nil {
-			util.Failed("There was a problem configuring your application: %v\n", err)
-		}
+		if siteName == "" && docrootRelPath == "" && pantheonEnvironment == "" && appType == "" {
+			err = c.PromptForConfig()
+			if err != nil {
+				util.Failed("There was a problem configuring your application: %v\n", err)
+			}
+		} else {
+			// Handle the case where flags have been passed in and we config that way
+			if appType == "" {
+				appType, err = ddevapp.DetermineAppType(c.Docroot)
+				if err != nil {
+					fullPath, _ := filepath.Abs(c.Docroot)
+					util.Failed("Failed to determine app type (drupal7/drupal8/wordpress).\nYour docroot %v may be incorrect - looking in directory %v, error=%v", c.Docroot, fullPath, err)
+				}
+			}
```

Comment:
We probably still want to validate appType against the docroot provided and give a warning if there is a mismatch.

Here's an example of what happened when I used "wordpress" as the apptype for a d7 site:
```
testing/drupal-kickstart ‹minikube› git:(master) » ddev config pantheon --pantheon-environment=dev --docroot=docroot --apptype=wordpress --sitename=d7kickstart
Found a drupal7 codebase at /Users/beeradb/Projects/testing/drupal-kickstart/docroot
Configuration complete. You may now run `ddev start` or `ddev pull`
Initial configuration file written successfully. See the configuration file for additional configuration options.
```

Reply:
This looks correct to me. If you specify an appType, it's explicitly overriding the auto-detect. What should it do instead of that?  

What kind of generic validation of the docroot would you recommend? 

Essentially the command-line is overrides "I know what I'm doing, shut up now".  If that's not what we want, something else can happen, but it looks to me like that's what you have here.

Reply:
An ideal solution would probably be to accept their value but warn them about a mismatch. i.e. some yellow warning text (or similar) that says something like:

"You specified WordPress as the site type, but this appears to be a Drupal site".

Even people that know what they are doing make mistakes! :)

Reply:
> What kind of generic validation of the docroot would you recommend?

Again, probably fine to accept the value, but a warning if the directory doesn't exist might be nice. Just in case someone makes a mistake.